index_impl.go

```diff
@@ -482,6 +483,18 @@ func (i *indexImpl) SearchInContext(ctx context.Context, req *SearchRequest) (sr
 	ctx = context.WithValue(ctx, search.SearchIOStatsCallbackKey,
 		search.SearchIOStatsCallbackFunc(sendBytesRead))
 
+	var bufPool *s2.GeoBufferPool
+	getBufferPool := func() *s2.GeoBufferPool {
+		if bufPool == nil {
+			bufPool = s2.NewGeoBufferPool(search.MaxBufPoolSize, search.MinBufPoolSize)
+		}
+
+		return bufPool
+	}
+
+	ctx = context.WithValue(ctx, search.GeoBufferPoolCallbackKey,
+		search.GeoBufferPoolCallbackFunc(getBufferPool))
```

Comment:
Why declare this here and send that in as context to retrieved by the geoshape searcher?

It seems we're not really sharing this across searchers either, so why not simply declare this where we need it?

Reply:
The buffer is declared in SearchInContext and attached to the context so that any searcher with the geo component can access this function. Only the first searcher will trigger the creation of the bufPool while the other calls will simply get the reference to this pool. I have tested this path using a conjunction query with 2 geoshape components and it seems to be working as intended